doc/fuzzing.md

```diff
@@ -87,11 +102,33 @@ To build all fuzz targets with libFuzzer, run
 make
 ```
 
-The fuzzer needs some inputs to work on, but the inputs or seeds can be used
-interchangeably between libFuzzer and AFL.
-
 See https://llvm.org/docs/LibFuzzer.html#running on how to run the libFuzzer
 instrumented executable.
 
 Alternatively run the script in `./test/fuzz/test_runner.py` and provide it
 with the `${DIR_FUZZ_IN}` created earlier.
+
+### macOS hints for libFuzzer
+
+The default clang/llvm version supplied by Apple on MacOS does not include
+fuzzing libraries, so macOS users will need to install a full version, for
+example using `brew install llvm`.
+
+When executing ./configure you should put --disable-asm to avoid errors
+with certain assembly code from Bitcoin Core's code.
+
+You may also need to take care of giving the correct path for clang and
+clang++, like `CC=/path/to/clang CXX=/path/to/clang++` if the non-systems
+clang does not come first in your path.
+
+If you run into problems with "boost sleep" or some of boost's libraries
+can't be found, like boost.thread or boost.filesystem, add this to your
+configure:
+```
+CXXFLAGS="-isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk"
```

Comment:
If this is needed, we should be using something like `$(xcrun --show-sdk-path)` (same as we do in [depends](https://github.com/bitcoin/bitcoin/blob/master/depends/builders/darwin.mk#L14)), rather than hard-coding a path to a specific SDK version.


Although I'm curious to know why this is would be required, as it's already added to the compiler flags by clang. i.e:

```bash
/usr/local/Cellar/llvm/9.0.0_1/bin/clang++ test.cpp -o test -v
...
"/usr/local/Cellar/llvm/9.0.0_1/bin/clang-9" 
-cc1 
-triple x86_64-apple-macosx10.14.0 
< snip a bunch of args> 
-target-linker-version 512.4 
-v -resource-dir /usr/local/Cellar/llvm/9.0.0_1/lib/clang/9.0.0 
-isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
``

Reply:
The **sysroot** there is pointing to /Library/Developer which is not correct when using under Catalina (but could be correct under previous macOS versions). Here a link to a question from SO regarding this particular problem: https://stackoverflow.com/questions/58278260/cant-compile-a-c-program-on-a-mac-after-upgrading-to-catalina-10-15

Reply:
> The sysroot there is pointing to /Library/Developer which is not correct when using under Catalina (but could be correct under previous macOS versions). 

@brakmic Sure, however that's why we'd want to use `xcrun --show-sdk-path`, because it should point to the right location for the SDK regardless of the OS it's being run on. What does it output when you run it on Catalina?

Reply:
Updated to use `xcrun --show-sdk-path`.